### PR TITLE
Frame times in title bar

### PR DESCRIPTION
**Pull request by BaronKiko**

### CODE REVIEW

**Comment by emmauss** [comment]
though this seems like a good idea, you have not given any description of what this is and how it would improve ryujinx.

**Comment by BaronKiko** [comment]
Sorry, it makes it easier to see fluctuations in performance. I implemented it to test how performance is effected for converting quads to triangles. Pr for that coming in a min.

**Comment by emmauss** [comment]
sounds great. could you join the discord server. Makes it easier to discuss this stuff

File: Ryujinx/Ui/GLScreen.cs [hunk 1]
```diff
@@ -246,14 +246,23 @@ private new void RenderFrame()
...
             _device.Statistics.RecordSystemFrameTime();
 
+            // System
             double hostFps = _device.Statistics.GetSystemFrameRate();
+            double hostFrameTime = _device.Statistics.GetSystemFrameTime();
+            double hostAvgFrameTime = _device.Statistics.GetSystemAverageFrameTime();
+
+            // Game
             double gameFps = _device.Statistics.GetGameFrameRate();
+            double gameFrameTime =_device.Statistics.GetGameFrameTime();
```
Comment (emmauss) [inline Ryujinx/Ui/GLScreen.cs]: these ones too

File: Ryujinx/Ui/GLScreen.cs [hunk 2]
```diff
@@ -246,14 +246,23 @@ private new void RenderFrame()
 
             _device.Statistics.RecordSystemFrameTime();
 
+            // System
             double hostFps = _device.Statistics.GetSystemFrameRate();
+            double hostFrameTime = _device.Statistics.GetSystemFrameTime();
+            double hostAvgFrameTime = _device.Statistics.GetSystemAverageFrameTime();
```
Comment (emmauss) [inline Ryujinx/Ui/GLScreen.cs]: align these

**Comment by BaronKiko** [comment]
Done

File: Ryujinx/Ui/GLScreen.cs [hunk 3]
```diff
@@ -246,14 +246,23 @@ private new void RenderFrame()
...
-            double hostFps = _device.Statistics.GetSystemFrameRate();
-            double gameFps = _device.Statistics.GetGameFrameRate();
+            // System
+            double hostFps           = _device.Statistics.GetSystemFrameRate();
+            double hostFrameTime     = _device.Statistics.GetSystemFrameTime();
+            double hostAvgFrameTime  = _device.Statistics.GetSystemAverageFrameTime();
+
+            // Game
+            double gameFps           = _device.Statistics.GetGameFrameRate();
+            double gameFrameTime     =_device.Statistics.GetGameFrameTime();
```
Comment (AcK77) [inline Ryujinx/Ui/GLScreen.cs]: You forgot a space after the =

Comment (BaronKiko) [reply Ryujinx/Ui/GLScreen.cs]: Once again done

**Comment by BaronKiko** [comment]
The averages is actually pretty useless because it's just 1000/fps. It's there to help you understand deviation in the immediate frame time (because doing 1000/fps in your head isn't exactly always obvious). The reason immediate frame time is so important is because it helps you measure frame consistency which is arguably more important to overall playablility than frame rate is.
For example if you have a game that runs at 60fps (16ms) you should never see the 100's digit become a 1 because it means some frames are taking over 5 times as long. Likewise if it's all over the place on a static menu then something is probably not right.
Ideally is should go in a bottom bar or something else you can disable but at there isn't any UI outside of the title bar it went there to avoid implementing lots of UI.
Tl;dr: I think it's all useful and needed but it should be moved off the title bar when some UI is in place to house it.

**Comment by BaronKiko** [comment]
Thinking about it, you could use a config option to hide it. You would probably want to disable by default for users but not when doing a regular build in visual studio.
Regardless I think that's all a bit out of scope for this pr so it should probably be done separately.

**Comment by BaronKiko** [comment]
Rip first commit.
I am moving this info to my new profiler instead: https://github.com/Ryujinx/Ryujinx/pull/567
